### PR TITLE
docs(sample): Add a top-level Datastore samples README

### LINKED ISSUE #745: What are the steps to run samples/snippets/src/main/java/com/google/datastore/snippets/TaskList.java ?
Thanks for stopping by to let us know something could be better!

**Is your feature request related to a problem? Please describe.**
What the problem is. Example: I'm unable to run TaskList.java.

**Describe the solution you'd like**
We previously documented the steps to run the sample [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/8a6966fbf71a9629997cd381f397460032768e78/datastore/README.md)

I'd like to see a similar README restored. Someone suggested that a pom.xml file might also be missing.

### PR DESCRIPTION
**Pull request by bw-yuhogg**
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea. **Issue already existed**
- [x] Ensure the tests and linter pass **N/A**
- [x] Code coverage does not decrease (if any source code was changed) **N/A**
- [x] Appropriate docs were updated (if necessary) **Strictly a docs addition**

Fixes #745 ☕️

Note: I can confirm that the existing pom.xml works correctly.

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Getting Started with Datastore Samples
+
+## Running a sample using Cloud Shell
+
+The Google Cloud Shell has application default credentials from its compute instance which will allow you to run samples without having to obtain `GOOGLE_APPLICATION_CREDENTIALS`.
+Go to the [Datastore Client Readme](https://github.com/googleapis/java-datastore#samples) to run each sample in the Cloud Shell.
+
+## Running a sample at the command line
+
+First set up `GOOGLE_APPLICATION_CREDENTIALS` and `GOOGLE_CLOUD_PROJECT` environment variables before running any samples.
```
Comment (lesv) [inline samples/README.md:10]: The shouldn't need to set `GOOGLE_APPLICATION_CREDENTIALS` if they use `gcloud auth application-default login`

**Review by lesv — APPROVED** [review]
Consider the comment

File: samples/README.md (hunk 1 above)
Comment (rwhogg) [reply samples/README.md]: Fixed, thanks!

**Review by kolea2 — APPROVED** [review]

**Review by kolea2 — CHANGES_REQUESTED** [review]
left one last comment on thread

**Review by kolea2 — APPROVED** [review]

**Pull request merged**